lib/cocoa/src/protocol/TCompactProtocol.m

```diff
@@ -0,0 +1,706 @@
+    if (size == 15)
+    {
+        size = (int)[self readVarint32];
+    }
+    
+    int elementType = [self getTTypeForCompactType:size_and_type & 0x0f];
+    
+    if (pelementType != NULL)
+    {
+        *pelementType = elementType;
+    }
+    if (psize != NULL)
+    {
+        *psize = size;
+    }
+}
+
+-(void)readSetBeginReturningElementType:(int*)pelementType size:(int*)psize
+{
+    [self readListBeginReturningElementType:pelementType size:psize];
+}
+
+-(BOOL)readBool
+{
+    if (booleanValue != nil)
+    {
+        BOOL result = [booleanValue boolValue];
+        [booleanValue release_stub];
+        booleanValue = nil;
+        return result;
+    }
+    else
+    {
+        return [self readByte] == TCType_BOOLEAN_TRUE;
+    }
+}
+
+-(uint8_t)readByte
+{
+    uint8_t buf = 0;
+    [mTransport readAll:&buf offset:0 length:1];
+    return buf;
+}
+
+-(short)readI16
+{
+    return (short)[self zigZagToi32:[self readVarint32]];
+}
+
+-(int)readI32
+{
+    return [self zigZagToi32:[self readVarint32]];
+}
+
+-(int64_t)readI64
+{
+    return [self zigZagToi64:[self readVarint64]];
+}
+
+-(double)readDouble
+{
+    uint64_t bits = 0;
+    [mTransport readAll:(uint8_t*)&bits offset:0 length:8];
+    bits = OSSwapLittleToHostInt64(bits);
+    
+    uint8_t data[8];
+    double result = 0;
+    memcpy(data, &bits, 8);
+    memcpy(&result, data, 8);
+    
+    return result;
+}
+
+-(NSString*)readString
+{
+    int length = (int)[self readVarint32];
+    if (length == 0)
+    {
+        return @"";
+    }
+    
+    return [[[NSString alloc] initWithData:[self readBinary:length] encoding:NSUTF8StringEncoding] autorelease_stub];
+}
+
+-(NSData*)readBinary
+{
+    return [self readBinary:(int)[self readVarint32]];
+}
+
+-(NSData*)readBinary:(int)length
+{
+    if (length == 0)
+    {
+        return [NSData data];
+    }
+    
+    NSMutableData* buf = [NSMutableData dataWithLength:length];
+    [mTransport readAll:buf.mutableBytes offset:0 length:length];
+    return buf;
+}
+
+-(void)readMessageEnd {}
+-(void)readFieldEnd {}
+-(void)readMapEnd {}
+-(void)readListEnd {}
+-(void)readSetEnd {}
+
+-(uint32_t)readVarint32
+{
+    uint32_t result = 0;
+    int shift = 0;
+    while (true)
+    {
+        uint8_t byte = [self readByte];
+        result |= (uint32_t)(byte & 0x7f) << shift;
+        if (!(byte & 0x80))
+        {
+            break;
+        }
+        shift += 7;
+    }
+    return result;
+}
+
+-(uint64_t)readVarint64
+{
+    int shift = 0;
+    uint64_t result = 0;
+    while (true)
+    {
+        uint8_t byte = [self readByte];
+        result |= (uint64_t)(byte & 0x7f) << shift;
+        if (!(byte & 0x80))
+        {
+            break;
+        }
+        shift += 7;
+    }
+    
+    return result;
+}
+
+-(int)zigZagToi32:(uint32_t)n
+{
+    return (int)(n >> 1) ^ (-(int)(n & 1));
+}
+
+-(int64_t)zigZagToi64:(uint64_t)n
+{
+    return (int64_t)(n >> 1) ^ (-(int64_t)(n & 1));
+}
+
+-(int)getTTypeForCompactType:(uint8_t)type
```

Comment:
This doesn't match the Cocoa naming guidelines. `get…` methods are only used for by-ref return values, like `-[NSData getBytes:range:]` which returns data byref in the `bytes` argument. It should be renamed to `ttypeForCompactType:`.